docs/adr/0007-seperate-binding-generators.md

```diff
@@ -0,0 +1,142 @@
+# Separate the binding generators into their own crates per target
+
+* Status: proposed
+* Deciders: TBD
+* Date: 2022-01-16
+
+Technical Story: [Issue 299](https://github.com/mozilla/uniffi-rs/issues/299)
+
+outdated early prototype: [PR 997](https://github.com/mozilla/uniffi-rs/pull/997)
+
+## Context and Problem Statement
+All the binding generators currently live in the [`uniffi_bindgen`](../../uniffi_bindgen/src/bindings) crate. This creates the following difficulties:
+- All the bindings live in the `uniffi` repository, so the `uniffi` team has to maintain them (or at the very least review changes to them).
```

Comment:
This may be a pro or a con, depending on how much the `uniffi` team likes maintaining the language in question 😆. Regardless, "separate crates" does not necessarily mean "separate repo" - do you intend to move e.g. the Kotlin bindings to their own repo as part of this work, perhaps to simplify CI?

Reply:
Not as a part of this work - but it'll be interesting to have the discussion in the future 😬 The upcoming firefox desktop bindings will probably be the first to live outside the repo 😄